main.tf

```
@@ -55,11 +55,12 @@ resource "google_storage_bucket" "main" {
   storage_class      = "REGIONAL"
   labels             = var.bucket_labels
   bucket_policy_only = true
+  count              = var.create_bucket == true ? 1 : 0
 }
 
 resource "google_storage_bucket_object" "main" {
   name                = "${data.archive_file.main.output_md5}-${basename(data.archive_file.main.output_path)}"
-  bucket              = google_storage_bucket.main.name
+  bucket              = var.create_bucket == true ? google_storage_bucket.main[0].name : var.bucket_name
```

```suggestion
  bucket              = var.create_bucket ? google_storage_bucket.main[0].name : var.bucket_name
```
nit: since var.create_bucket is a `bool` I believe this should be possible

---

adopted